XamarinCommunityToolkit/Behaviors/ValidationBehavior.shared.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System.ComponentModel;
+using System.Windows.Input;
+using Xamarin.Forms;
+
+namespace XamarinCommunityToolkit.Behaviors
+{
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public abstract class ValidationBehavior : BaseBehavior
+    {
+        public static readonly BindableProperty IsValidProperty = BindableProperty.Create(nameof(IsValid), typeof(bool), typeof(ValidationBehavior), true, BindingMode.OneWayToSource);
+
+        public static readonly BindableProperty ValidStyleProperty = BindableProperty.Create(nameof(ValidStyle), typeof(Style), typeof(ValidationBehavior), propertyChanged: OnValidationPropertyChanged);
+
+        public static readonly BindableProperty InvalidStyleProperty = BindableProperty.Create(nameof(InvalidStyle), typeof(Style), typeof(ValidationBehavior), propertyChanged: OnValidationPropertyChanged);
+
+        public static readonly BindableProperty FlagsProperty = BindableProperty.Create(nameof(Flags), typeof(ValidationFlags), typeof(ValidationBehavior), ValidationFlags.ValidateOnUnfocusing | ValidationFlags.ForceMakeValidWhenFocused, propertyChanged: OnValidationPropertyChanged);
+
+        public static readonly BindableProperty ValueProperty = BindableProperty.Create(nameof(Value), typeof(object), typeof(ValidationBehavior), propertyChanged: OnValuePropertyChanged);
+
+        public static readonly BindableProperty ValuePropertyNameProperty = BindableProperty.Create(nameof(ValuePropertyName), typeof(string), typeof(ValidationBehavior), defaultValueCreator: GetDefaultValuePropertyName, propertyChanged: OnValuePropertyNamePropertyChanged);
+
+        public static readonly BindableProperty ForceValidateCommandProperty = BindableProperty.Create(nameof(ForceValidateCommand), typeof(ICommand), typeof(ValidationBehavior), defaultValueCreator: GetDefaultForceValidateCommand, defaultBindingMode: BindingMode.OneWayToSource);
```

Comment:
I'll stop mentioning it from here, there might be more, but again some breaks :)